config/7.1.0/securebanking/ig/scripts/groovy/VRPTypeVerifier.groovy

```diff
@@ -0,0 +1,36 @@
+import org.forgerock.http.protocol.*
+import com.forgerock.sapi.gateway.rest.HttpHeaderNames
+
+String fapiInteractionId = request.getHeaders().getFirst(HttpHeaderNames.X_FAPI_INTERACTION_ID)
+if (fapiInteractionId == null) {
+    fapiInteractionId = 'No ' + HttpHeaderNames.X_FAPI_INTERACTION_ID + ' header'
+}
+SCRIPT_NAME = '[VRPTypeVerifier] (' + fapiInteractionId + ') - '
+logger.debug(SCRIPT_NAME + 'Running...')
+
+def vrpTypeFromRequest = request.entity.getJson().Data.ControlParameters.VRPType
+logger.debug(SCRIPT_NAME + 'ControlParameters' + request.entity.getJson().Data.ControlParameters)
+logger.debug(SCRIPT_NAME + 'vrpTypeFromRequest: ' + vrpTypeFromRequest)
+
+def vrpAllowed = false
+
+for (String vrpType : vrpTypeFromRequest) {
+    logger.debug(SCRIPT_NAME + 'vrpType: ' + vrpType)
+    if (vrpType == 'UK.OBIE.VRPType.Sweeping') {
+        vrpAllowed = true;
+        break;
+    }
+}
```

Comment:
If the platform support only `UK.OBIE.VRPType.Sweeping` we need to reject the consent if any other VRPtype is provided in that list, for example:
VRPType = ["`UK.OBIE.VRPType.Sweeping`", "`UK.OBIE.VRPType.Other`"] reject the consent because we support only Sweeping.

In short the list need to contains only `UK.OBIE.VRPType.Sweeping`
VRPType = ["`UK.OBIE.VRPType.Sweeping`"]

Reply:
Probably the for could be deleted and use a simple conditional.